### ISSUE TITLE
Blufi example init and deinit memory leak (IDFGH-14055)

### ISSUE THREAD

**Issue by josthlee** [espressif/esp-idf#14874]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.4-dev-4076-gce6085349f

### Espressif SoC revision.

ESP32-C6 (QFN40) (revision v0.0)

### Operating System used.

Linux

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

None

### Power Supply used.

USB

### What is the expected behavior?

TL;DR is that I use esp_blufi_host_init and esp_blufi_host_deinit to start and stop NIMBLE blufi with CONFIG_BT_NIMBLE_ENABLED. However, it leaks 272 bytes every time with an init and deinit. 

### What is the actual behavior?

The root cause is the os_msys_get in esp_blufi.c line [226](https://github.com/espressif/esp-idf/blob/master/components/bt/common/btc/profile/esp/blufi/nimble_host/esp_blufi.c#L226) and [234](https://github.com/espressif/esp-idf/blob/master/components/bt/common/btc/profile/esp/blufi/nimble_host/esp_blufi.c#L234) and it never frees the allocated memory in deinit function.

### Steps to reproduce.

I just add a workaround myself in components/bt/common/btc/profile/esp/blufi/nimble_host/esp_blufi.c:: bluesp_blufi_btc_deinit since the variables are file scoped in esp_blufi.c and I hope you can patch it in a proper place.

```
void esp_blufi_btc_deinit(void)
{           
                
    int i = 0;  
    const struct ble_gatt_svc_def *svc;
    const struct ble_gatt_chr_def *chr;
    const struct ble_gatt_dsc_def *dsc;
                
    for (svc = gatt_svr_svcs; svc && svc->uuid; svc++) {
        for (chr = svc->characteristics; chr && chr->uuid; chr++) {
            os_mbuf_free(gatt_values[i].buf);
            ++i;
            
            for (dsc = chr->descriptors; dsc && dsc->uuid; dsc++) {
                os_mbuf_free(gatt_values[i].buf);
                ++i;
            }
        }       
    }           
            
    btc_deinit();
} 
```

**Comment by strange969** [comment]
Hi @josthlee,

Thank you for bringing up this issue. I have looked into it and implemented the necessary code for deinitialization. I’m attaching the patch files for the IDF and NimBLE changes.

Please apply these patches to your repository. If you encounter any issues or need further assistance, feel free to let me know.

[blufi_memory_leak_changes.zip](https://github.com/user-attachments/files/17813925/blufi_memory_leak_changes.zip)

**Comment by josthlee** [comment]
> Hi @josthlee,
> 
> Thank you for bringing up this issue. I have looked into it and implemented the necessary code for deinitialization. I’m attaching the patch files for the IDF and NimBLE changes.
> 
> Please apply these patches to your repository. If you encounter any issues or need further assistance, feel free to let me know.
> 
> [blufi_memory_leak_changes.zip](https://github.com/user-attachments/files/17813925/blufi_memory_leak_changes.zip)

Noted and thank you. I will try the patch and then get back to you!

**Comment by josthlee** [comment]
Thank you and I think I don't have any issue on this now. Will then close the issue, thanks!
